### ISSUE TITLE
[Bug Report] Transformer Lens from_pretrained weights not matching HuggingFace model weights

### ISSUE THREAD

**Issue by msakarvadia** [neelnanda-io/TransformerLens#526]
**When using `from_pretrained` to load model, the weights don't match their HuggingFace counterparts**

I tried to verify the the embedding matrix for the `EleutherAI/gpt-neo-125M` model loaded via transformer-lens, matched that of a model loaded via HuggingFace directly. I observed that the embeddings do not match. However, I notice in the `from_pretained` method they should be equivalent: https://github.com/neelnanda-io/TransformerLens/blob/main/transformer_lens/loading_from_pretrained.py#L1469 

**Code example**
```
from transformers import AutoModelForCausalLM
!pip install transformer_lens #version 1.14.0
import transformer_lens
from transformer_lens import HookedTransformer

model_name = "EleutherAI/gpt-neo-125M"
model_tl = HookedTransformer.from_pretrained(model_name)
tl_embed = model_tl.W_E
model_hf = AutoModelForCausalLM.from_pretrained(model_name)
hf_embed = model_hf.transformer.wte.weight

# hf_embed != tl_embed (even though these two should be equivalent)

See this colab notebook for an error replication: https://colab.research.google.com/drive/1-uBJWtATmhc-yPsbGM4-_cQmBohgsgB1?usp=sharing 

```

**System Info**
transformer_lens version 1.14.0
transformers version 4.38.2

**Additional Context**
It is possible that this weight mismatch might be intentional, and I am simply missing a peice of knowledge. Feel free to let me know if this is the case with an explanation of why there is a weight mismatch. Thanks in advance!

### Checklist

- [ ] I have checked that there is no similar [issue](https://github.com/neelnanda-io/TransformerLens/issues) in the repo (**required**)

**Comment by neelnanda-io** [comment]
`from_pretrained` applies various forms of post-processing on the weights,
like making their output mean zero (in models with LayerNorm) and folding
in LayerNorm weights. This is intended to make interpretability cleaner,
while keeping things mathematically equivalent, ie without changing the
final log probs (almost all changes also preserve the logits, apart from
ensuring the unembed is mean zero which adds a constant amount to each
logit). This should all be documented in the docstring.

Use `from_pretrained_no_processing` if you want to disable this (there are
also boolean flags for whether to toggle each form of post-processing on or
off)

**Comment by msakarvadia** [comment]
Thanks for pointing that out, makes a lot of sense now!
